flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -405,20 +395,37 @@ public void testIdleAndBackPressuredTime() throws IOException, InterruptedExcept
         // wait until request thread start to run.
         syncLock.await();
 
-        Thread.sleep(100);
+        // wait until request buffer blocking.
+        boolean bufferPoolBlocking = false;
+
+        for (int i = 0; i < 50; i++) {
+            StackTraceElement[] stackTrace = requestThread.getStackTrace();
+            bufferPoolBlocking = isInBlockingBufferRequest(stackTrace);
+
+            if (bufferPoolBlocking) {
+                break;
+            } else {
+                // Retry
+                Thread.sleep(500);
+            }
+        }
+
+        // Verify that Thread was in blocking request
+        assertThat(bufferPoolBlocking)
+                .isTrue()
+                .withFailMessage("Did not trigger blocking buffer request.");
```

Comment:
```suggestion
        while (!isInBlockingBufferRequest(requestThread.getStackTrace()) {
                Thread.sleep(50);
         }
```

Reply:
Thanks for your review @zentol, we must add timeout to this test if it is modified like your suggestion, right?